utils/benchmark.py

```diff
@@ -101,12 +101,21 @@ def _dump_results(self):
         with self._dump_filelock:
             with open(self._dump_filepath, "w") as f:
                 times_invoked = self._times_invoked
-                json.dump({
+                json_data = {}
+                json_data["overall"] = {
                     "workload_size": self._workload_size[self.warm_up_runs:times_invoked],
                     "start_times": self._start_times[self.warm_up_runs:times_invoked],
                     "finish_times": self._finish_times[self.warm_up_runs:times_invoked],
                     "threads": os.environ.get("AIO_NUMA_CPUS")
-                }, f)
+                }
+                for subcategory in self._subcategories.keys():
+                    json_data[subcategory] = {
+                        "workload_size": self._subcategories[subcategory]["workload_size"][self.warm_up_runs:times_invoked], # noqa
+                        "start_times": self._subcategories[subcategory]["start_times"][self.warm_up_runs:times_invoked],
```

Comment:
Here might be the source of the issue I've mentioned on mzd side. You cut subcategories' runs based on times_invoked as seen from outside of the llama generative loop.

Reply:
The problem was actually discarding warmup runs when they were already ignored by `start_subcategory_measurement` and `finish_subcategory_measurement`, which resulted in not meeting the minimum measurements count